components/brave_wallet/browser/polkadot/polkadot_substrate_rpc.cc

```diff
@@ -432,6 +432,54 @@ void PolkadotSubstrateRpc::OnGetBlockHeader(GetBlockHeaderCallback callback,
   std::move(callback).Run(std::move(*header), std::nullopt);
 }
 
+void PolkadotSubstrateRpc::GetBlockHash(std::string_view chain_id,
+                                        std::optional<uint32_t> block_number,
+                                        GetBlockHashCallback callback) {
+  auto url = GetNetworkURL(chain_id);
+
+  base::ListValue params;
+
+  if (block_number) {
+    params.Append(static_cast<double>(*block_number));
```

Comment:
What is the possible range of `block_number`. We should avoid a) `static_cast` and b) `double`

Reply:
Actually it seems hex should work 
https://github.com/w3f/PSPs/blob/b6d570173146e7a012cf43d270177e02ed886e2e/PSPs/drafts/psp-6.md#1941-parameter